pkg/secrets-store/nodeserver.go

```diff
@@ -151,6 +116,33 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 		log.Infof("skipping calling provider as it's mock")
 		return &csi.NodePublishVolumeResponse{}, nil
 	}
+
+	/// TODO: providerName is here for backward compatibility. Will eventually deprecate.
```

Comment:
we can remove this comment if we want to remove support for `providerName` as part of the`volumeAttributes` of the pod spec. then make sure we add a warning about it in the release note.

Reply:
Opened an issue to track this - https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/233